### PR TITLE
Update github.com/rubenv/sql-migrate

### LINKED ISSUE #9770: Drop import on github.com/go-kit/kit
This dependency imports ~the entire golang ecosystem, and thus makes it hard to import helm.

See https://github.com/prometheus/common/issues/255 for a lot more context.

### PR DESCRIPTION
**Pull request by howardjohn**
**What this PR does / why we need it**:
Updates the dependency on github.com/rubenv/sql-migrate. This drops the total LOC imported by this library from 13.7M to 11.9M due to dropping and import on github.com/go-kit/kit.
Fixes https://github.com/helm/helm/issues/9770

See https://github.com/prometheus/common/issues/255 for more information on why this may be important
**Special notes for your reviewer**:
Note: for this dependency there are no tags, so `master` was used. According to the author in https://github.com/rubenv/sql-migrate/issues/158, there will never be a breaking change to this library, so it should be safe to update.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by cndoit18 — COMMENTED** [review]
Hello
PRs must be linked to an existing issue. If none exists, may you please open one explaining the issue and expected outcome?
Please also write tests to test and verify this PR fixes your issue. Thanks.

**Comment by howardjohn** [comment]
@cndoit18 I opened https://github.com/helm/helm/issues/9770 which this fixes. I am not sure its relevant to add a test, this is not a behavioral issue but rather an issue that we are importing too many things.

**Comment by howardjohn** [comment]
Any feedback on what is needed to get this merged?

**Comment by bacongobbler** [comment]
> Note: for this dependency there are no tags, so `master` was used. According to the author in [rubenv/sql-migrate#158](https://github.com/rubenv/sql-migrate/issues/158), there will never be a breaking change to this library, so it should be safe to update.

While I understand the sentiment, we have been burned by similar arguments made in the past 😄

We will have to check the changes made upstream to check and make sure.

Thanks for being patient.

**Comment by howardjohn** [comment]
I would encourage you to reconsider the priority, as the change is quite small and the impact of this change extends beyond Helm, as you are importing a gigantic library, any consumers of helm as a library are also required to import it, and all importers of that library, and so on.

A considerable effort across the golang ecosystem has been made to fix this bloated dependency problem, but it requires buy-in from everyone.

File: go.mod
Comment (howardjohn) [inline go.mod:16]: I didn't intentionally remove this, `go mod tidy` wants to remove it

**Review by bacongobbler — APPROVED** [review]
should be good to go after a rebase.

**Comment by howardjohn** [comment]
@bacongobbler thanks,  should be good to go now

**Comment by bacongobbler** [comment]
Sorry, forgot to mention yesterday. Due to the code size, as per policy this will need another maintainer's sign-off before we can merge.

https://github.com/helm/helm/blob/main/CONTRIBUTING.md#size-labels

**Comment by howardjohn** [comment]
> Sorry, forgot to mention yesterday. Due to the code size, as per policy this will need another maintainer's sign-off before we can merge.
> 
> https://github.com/helm/helm/blob/main/CONTRIBUTING.md#size-labels

Fair enough, although the doc does say ignoring generated code - the change is 2 lines of real change and the rest is generated

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @howardjohn

**Pull request merged**
